### PR TITLE
CTOR-1051 : Refresh cloud-aws-cloudfront

### PR DESCRIPTION
**Pull request by lucie-dubrunfaut**
## Description

Refresh cloud-aws-cloudfront

## Target version (i.e. version that this PR changes)

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] 24.10.x
- [ ] Cloud
- [x] Monitoring Connectors

### CODE REVIEW

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 1]
```diff
@@ -5,56 +5,77 @@ title: Amazon CloudFront
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
+## Dépendances du Connecteur de supervision
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:8]:
```suggestion
## Dépendances du connecteur de supervision
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 2]
```diff
@@ -63,18 +84,15 @@ Vous trouverez plus d'informations sur la découverte d'Hôtes et son fonctionne
...
-
-| AWS Privilege                  | Description                                                     |
-| :----------------------------- | :-------------------------------------------------------------- |
-| cloudwatch:getMetricStatistics | Get metrics from the AWS/EC2 namespace on Cloudwatch.           |
-| cloudfront:ListDistributions   | Grants permission to list the distributions associated with your AWS account. |
+Configurez un compte de service (via une combinaison d'access et secret key) et affectez-lui les privilèges suivants :
+* cloudwatch:getMetricStatistics
+* cloudfront:ListDistributions
 
 ### Dépendances du Plugin
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:91]:
```suggestion
### Dépendances du plugin
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 3]
```diff
@@ -165,85 +189,267 @@ yum install centreon-plugin-Cloud-Aws-Cloudfront-Api
...
+	--critical-bytes-downloaded='' \
+	--per-sec \
+	--verbose
 ```
 
 La commande devrait retourner un message de sortie similaire à :
 
 ```bash
 OK: Bytes Downloaded Bytes Uploaded | 'cloudfront.bytes.downloaded.bytes'=9000;;;; 'cloudfront.bytes.uploaded.bytes'=9000B;;;; 
+
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:313]:
```suggestion
```

File: pp/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 4]
```diff
@@ -5,56 +5,76 @@ title: Amazon CloudFront
...
+The connector brings the following service templates (sorted by the host template they are attached to):
 
-* Cloud-Aws-CloudFront-custom
+<Tabs groupId="sync">
+<TabItem value="Cloud-Aws-CloudFront-custom" label="Cloud-Aws-CloudFront-custom">
 
-It brings the following service templates:
+| Service Alias         | Service Template                           | Service Description                   |
+|:----------------------|:-------------------------------------------|:--------------------------------------|
+| Cloudfront-Errors     | Cloud-Aws-Cloudfront-Errors-Api-custom     | Check pages access errors             |
```
Comment (cg-tw) [inline pp/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:29]:
```suggestion
| Cloudfront-Errors     | Cloud-Aws-Cloudfront-Errors-Api-custom     | Check page access errors             |
```

File: pp/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 5]
```diff
@@ -5,56 +5,76 @@ title: Amazon CloudFront
...
+| Cloudfront-Requests   | Cloud-Aws-Cloudfront-Requests-Api-custom   | Check requests number                 |
```
Comment (cg-tw) [inline pp/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:30]:
```suggestion
| Cloudfront-Requests   | Cloud-Aws-Cloudfront-Requests-Api-custom   | Check the number of requests                 |
```

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 6]
```diff
@@ -165,57 +189,121 @@ yum install centreon-plugin-Cloud-Aws-Cloudfront-Api
...
-    --critical-bytes-downloaded='' \
-    --per-sec \
-    --verbose \
-    --use-new-perfdata
+/usr/lib/centreon/plugins/centreon_aws_cloudfront_api.pl \
+	--plugin=cloud::aws::cloudfront::plugin \
+	--mode=throughput \
+	--custommode='paws' \
+	--aws-secret-key='' \
+	--aws-access-key='' \
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:292]: ?

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 7]
```diff
@@ -165,57 +189,121 @@ yum install centreon-plugin-Cloud-Aws-Cloudfront-Api
...
+	--aws-role-arn='' \
+	--region='' \
+	--id='' \
```
Comment (cg-tw) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:295]: ?

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 8]
```diff
@@ -165,57 +189,121 @@ yum install centreon-plugin-Cloud-Aws-Cloudfront-Api
...
+| AWSASSUMEROLE | Set Amazon Resource Name of the role to be assumed                                                                                                 |                   |             |
+| AWSCUSTOMMODE | When a plugin offers several ways (CLI, library, etc.) to get information the desired one must be defined with this option                         | paws              |             |
+| AWSINSTANCEID | Set the instance ID (required) (can be defined multiple times)                                                                                     |                   | X           |
+| AWSREGION     | Set the region name (required)                                                                                                                     |                   | X           |
+| AWSSECRETKEY  | Set AWS secret key                                                                                                                                 |                   |             |
```
Comment (lucie-dubrunfaut) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:218]:
```suggestion
| AWSACCESSKEY  | Set AWS access key                                                                                                                                 |                   | X           |
| AWSASSUMEROLE | Set Amazon Resource Name of the role to be assumed                                                                                                 |                   |             |
| AWSCUSTOMMODE | When a plugin offers several ways (CLI, library, etc.) to get information the desired one must be defined with this option                         | paws              |             |
| AWSINSTANCEID | Set the instance ID (required) (can be defined multiple times)                                                                                     |                   | X           |
| AWSREGION     | Set the region name (required)                                                                                                                     |                   | X           |
| AWSSECRETKEY  | Set AWS secret key                                                                                                                                 |                   | X           |
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md (hunk 6 above)
Comment (lucie-dubrunfaut) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:292]:
```suggestion
	--aws-secret-key='XXXX' \
	--aws-access-key='XXXX' \
```

File: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md (hunk 7 above)
Comment (lucie-dubrunfaut) [inline i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:295]:
```suggestion
	--id='XXXX' \
```

File: pp/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 9]
```diff
@@ -166,85 +190,265 @@ yum install centreon-plugin-Cloud-Aws-Cloudfront-Api
...
+| AWSACCESSKEY  | Set AWS access key                                                                                                                       |                   |             |
+| AWSASSUMEROLE | Set Amazon Resource Name of the role to be assumed                                                                                       |                   |             |
+| AWSCUSTOMMODE | When a plugin offers several ways (CLI, library, etc.) to get information the desired one must be defined with this option               | paws              |             |
+| AWSINSTANCEID | Set the instance ID (required) (can be defined multiple times)                                                                           |                   | X           |
+| AWSREGION     | Set the region name (required)                                                                                                           |                   | X           |
+| AWSSECRETKEY  | Set AWS secret key                                                                                                                       |                   |             |
```
Comment (lucie-dubrunfaut) [inline pp/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:219]:
```suggestion
| AWSACCESSKEY  | Set AWS access key                                                                                                                       |                   | X           |
| AWSASSUMEROLE | Set Amazon Resource Name of the role to be assumed                                                                                       |                   |             |
| AWSCUSTOMMODE | When a plugin offers several ways (CLI, library, etc.) to get information the desired one must be defined with this option               | paws              |             |
| AWSINSTANCEID | Set the instance ID (required) (can be defined multiple times)                                                                           |                   | X           |
| AWSREGION     | Set the region name (required)                                                                                                           |                   | X           |
| AWSSECRETKEY  | Set AWS secret key                                                                                                                       |                   | X           |
```

File: pp/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md [hunk 10]
```diff
@@ -166,85 +190,265 @@ yum install centreon-plugin-Cloud-Aws-Cloudfront-Api
...
-    --use-new-perfdata
+/usr/lib/centreon/plugins/centreon_aws_cloudfront_api.pl \
+	--plugin=cloud::aws::cloudfront::plugin \
+	--mode=throughput \
+	--custommode='paws' \
+	--aws-secret-key='' \
+	--aws-access-key='' \
+	--aws-role-arn='' \
+	--region='' \
+	--id='' \
```
Comment (lucie-dubrunfaut) [inline pp/integrations/plugin-packs/procedures/cloud-aws-cloudfront.md:294]:
```suggestion
	--aws-secret-key='XXXX' \
	--aws-access-key='XXXX' \
	--aws-role-arn='' \
	--region='' \
	--id='XXXX' \
```

**Review by cg-tw — APPROVED** [review]

**Pull request merged**